framework/tlvf/src/include/tlvf/CmduParser.h

```diff
@@ -0,0 +1,35 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * Copyright (c) 2016-2019 Intel Corporation
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+#ifndef _CmduParser_H_
```

Comment:
I'm not sure if CmduParser is the right name. It is not just for CMDUs, it's for any type,length,value-structured sequence.

So I'd rather call it TLVListParser, and also create a TLVList class (from which CmduMessage derives) to collect all the functions that manipulate lists of TLVs.